src/screens/WalletConnect/CallRequest/SignatureRequestContent.js

```diff
@@ -55,11 +60,53 @@ function SignatureRequestContent({ request, onConfirm, onReject }: Props) {
   const { t } = useTranslation();
   const configs = useChainsConfig();
   const activeAccount = useActiveAccount();
+  const colors = useThemeColors();
   const isArchanovaAccountDeployed = useRootSelector(isArchanovaAccountDeployedSelector);
 
-  const { title, iconUrl, chain } = getViewData(request);
+  const { title, iconUrl, chain, method, params } = getViewData(request);
   const config = configs[chain];
 
+  const isSignTypedData = method === ETH_SIGN_TYPED_DATA || method === ETH_SIGN_TYPED_DATA_V4;
+
+  const { message, primaryType } =
+    isSignTypedData && !isEmpty(params) ? JSON.parse(params[1]) : { message: null, primaryType: null };
+
+  const messageText = React.useMemo(() => {
+    if (isEmpty(message)) return [];
+
+    const values = Object.entries(message);
+    if (isEmpty(values)) return [];
+
+    return values.map((value) => {
+      const key = value?.[0];
+      const description = value?.[1];
+
+      if (!key) return '';
+
+      const label = key.charAt(0).toUpperCase() + key.slice(1);
```

Comment:
if `label` is a `string`, then use string concatenation via template strings instead:

```suggestion
      const label = `${key.charAt(0).toUpperCase()}${key.slice(1)}`;
```

Reply:
Updated it!